content/en/docs/ambient/usage/extend-waypoint-wasm/wasm-module-distribution/index.md

```diff
@@ -0,0 +1,231 @@
+
+### Verify the traffic via the Gateway
+
+1. Test `/productpage` without credentials
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null "http://$GATEWAY_HOST/productpage"
+    401
+    {{< /text >}}
+
+1. Test `/productpage` with credentials
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -o /dev/null -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" -w "%{http_code}" "http://$GATEWAY_HOST/productpage"
+    200
+    {{< /text >}}
+
+## Apply Wasm Configuration at Waypoint Proxy
+
+### Deploy a waypoint proxy
+
+Follow the [waypoint deployment instructions](docs/ambient/getting-started/#layer-7-authorization-policy) to deploy a waypoint proxy in the bookinfo namespace.
+
+{{< text bash >}}
+$ istioctl x waypoint apply --enroll-namespace --wait
+{{< /text >}}
+
+### Verify traffic without WasmPlugin at the waypoint
+
+{{< text bash >}}
+$ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null http://productpage:9080/productpage
+200
+{{< /text >}}
+
+### Apply WasmPlugin at waypoint proxy
+
+To configure a WebAssembly filter with a remote Wasm module, create a `WasmPlugin` resource targeting the `waypoint` gateway:
+
+{{< text bash >}}
+$ kubectl get gateway
+NAME               CLASS            ADDRESS        PROGRAMMED   AGE
+bookinfo-gateway   istio            172.18.7.110   True         23h
+waypoint           istio-waypoint   10.96.202.82   True         21h
+{{< /text >}}
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: extensions.istio.io/v1alpha1
+kind: WasmPlugin
+metadata:
+  name: basic-auth-at-waypoint
+spec:
+  targetRefs:
+    - kind: Gateway
+      group: gateway.networking.k8s.io
+      name: waypoint # gateway name retrieved from previous step
+  url: oci://ghcr.io/istio-ecosystem/wasm-extensions/basic_auth:1.12.0
+  phase: AUTHN
+  pluginConfig:
+    basic_auth_rules:
+      - prefix: "/productpage"
+        request_methods:
+          - "GET"
+          - "POST"
+        credentials:
+          - "ok:test"
+          - "YWRtaW4zOmFkbWluMw=="
+EOF
+{{< /text >}}
+
+### View the configured WasmPlugin
+
+{{< text bash >}}
+$ kubectl get wasmplugin
+NAME                     AGE
+basic-auth-at-gateway    28m
+basic-auth-at-waypoint   14m
+{{< /text >}}
+
+### Verify the traffic via waypoint proxy
+
+1. Test internal `/productpage` without credentials
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null http://productpage:9080/productpage
+    401
+    {{< /text >}}
+
+1. Test internal `/productpage` with credentials
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" http://productpage:9080/productpage
+    200
```

Comment:
In this case, when you reach from productpage to reviews, does it also require the basic auth token?

Reply:
i guess not because prefix doesn't match

Reply:
> In this case, when you reach from productpage to reviews, does it also require the basic auth token?

yes, that's correct